cassandra/policies.py

```diff
@@ -839,6 +839,31 @@ def on_unavailable(self, query, consistency, required_replicas, alive_replicas,
         """
         return (self.RETRY_NEXT_HOST, None) if retry_num == 0 else (self.RETHROW, None)
 
+    def on_request_error(self, query, consistency, error, retry_num):
+        """
+        This is called when an unexpected error happens. This can be in the
+        following situations:
+
+        * On a connection error
+        * On server errors: overloaded, isBootstrapping, serverError, etc.
```

Comment:
How `OperationTimedOut` are handled? will the `RetryPolicy` involved in the decision?

Reply:
No. In our driver, we only have a global OperationTimedOut, so it is raised when that limit is reach (default 10s)

Reply:
ah, makes sense.